src/app/api/_utils/slack.ts

```diff
@@ -24,22 +24,34 @@ export function getSlackMessage(responseType: SlackResponseType, responseText: s
 }
 
 async function fetchSlackApi(slackApiString: string, team_id: string) {
-  // fetch the organization:
+  // Fetch the organization:
   const organization = await PostgresClient.getOrganization(team_id);
   const token = decrypt(organization.access_hash);
 
-  // const res = await fetch('https://slack.com/api/usergroups.users.list?usergroup=' + userGroupId,
-  const res = await fetch('https://slack.com/api/' + slackApiString,
-    {
+  try {
+    // const res = await fetch('https://slack.com/api/usergroups.users.list?usergroup=' + userGroupId,
+    const res = await fetch('https://slack.com/api/' + slackApiString, {
       method: 'GET',
       headers: {
-        'Authorization': 'Bearer ' + token, // TODO: we need to take the token we stored upon installation, not this one.
+        'Authorization': 'Bearer ' + token, // TODO: retrieve the correct token stored upon installation.
       },
+    });
+
+    const jsonResponse = await res.json();
+
+    if (jsonResponse.error === 'invalid_auth') {
+      throw new Error(`Error fetching from Slack API: ${jsonResponse.error || 'Unknown error'}.\nPlease check if the Slack token is valid. Reinstalling the app might resolve this issue.`);
     }
-  );
 
-  // if this fails, we must throw an error here, and handle it in the routes. something like: "the token is expired, please reinstall the app"
-  return await res.json();
+    if (!res.ok || jsonResponse.error) {
+      throw new Error(`Error fetching from Slack API: ${jsonResponse.error || 'Unknown error'}`);
+    }
+
+    return jsonResponse;
+  } catch (error) {
+    console.error(`Failed to fetch from Slack API: ${error}`);
+    throw error;
+  }
```

Comment:
same here, we don't even need the trycatch, if something really fails in this function, it should be handled by the route